### DISCUSSION TITLE
How to register custom FileProcessor in own rector project?

### DISCUSSION THREAD

**Discussion by simonschaufi** [rectorphp/rector#8267] [Q&A] [answered]
I'm wondering how to register a custom FileProcessor in my own rector project as I couldn't even figure out how the native `NonPhpFileProcessor` is registered. Just implementing the `FileProcessorInterface` doesn't do anything.

**Comment by TomasVotruba** [comment, marked as answer]
This would require implementing own tool, as Rector refocused directly on PHP files to meet expectations of other PHP-tooling:
https://github.com/rectorphp/rector-src/pull/4761
